src/editors/editorsController.ts

```diff
@@ -229,117 +232,126 @@ export default class EditorsController {
     });
   }
 
-  saveDocumentToMongoDB(): Promise<boolean> {
-    return new Promise(async (resolve) => {
-      const activeEditor = vscode.window.activeTextEditor;
-
-      log.info('save document to MongoDB', activeEditor);
+  _saveDocumentFailed(errorMessage: string) {
+    // Send a telemetry event that saving the document failed.
+    this._telemetryController.trackDocumentUpdated('treeview', false);
 
-      if (!activeEditor) {
-        return resolve(true);
-      }
+    vscode.window.showErrorMessage(errorMessage);
+  }
 
-      const uriParams = new URLSearchParams(activeEditor.document.uri.query);
-      const documentLocation =
-        uriParams.get(DOCUMENT_LOCATION_URI_IDENTIFIER) || '';
-      const namespace = uriParams.get(NAMESPACE_URI_IDENTIFIER) || '';
-      const connectionId = uriParams.get(CONNECTION_ID_URI_IDENTIFIER);
-      const documentIdReference =
-        uriParams.get(DOCUMENT_ID_URI_IDENTIFIER) || '';
-      const documentId = this._documentIdStore.get(documentIdReference);
+  async replaceDocument(
+    documentId: EJSON.SerializableTypes,
+    namespace: string,
+    dataservice: DataServiceType,
+    editor: vscode.TextEditor
+  ): Promise<boolean> {
+    let newDocument: EJSON.SerializableTypes = {};
 
-      if (
-        documentLocation !== 'mongodb' ||
-        !namespace ||
-        !connectionId ||
-        !documentId
-      ) {
-        vscode.commands.executeCommand('workbench.action.files.save');
+    try {
+      newDocument = EJSON.parse(editor.document.getText());
+    } catch (error) {
+      this._saveDocumentFailed(error.message);
 
-        return resolve(true);
-      }
+      return false;
+    }
 
-      const activeConnectionId = this._connectionController.getActiveConnectionId();
-      const connectionName = this._connectionController.getSavedConnectionName(
-        connectionId
-      );
+    this._statusView.showMessage('Saving document...');
 
-      if (activeConnectionId !== connectionId) {
-        // Send metrics to Segment.
-        this._telemetryController.trackDocumentUpdated('treeview', false);
+    const findOneAndReplace = util.promisify(dataservice.findOneAndReplace);
 
-        vscode.window.showErrorMessage(
-          `Unable to save document: no longer connected to '${connectionName}'`
-        );
+    try {
+      await findOneAndReplace(
```

Comment:
Isn't it throws an error: `Unable to save document: Cannot read property 'client' of undefined`?

Reply:
Ah yup needs a bind. Updating...